packages/api/internal/handlers/template_start_build.go

```diff
@@ -88,93 +90,88 @@ func (a *APIStore) PostTemplatesTemplateIDBuildsBuildID(c *gin.Context, template
+		build.RAMMB,
+	)
+
+	if buildErr != nil {
+		buildErr = fmt.Errorf("error when building env: %w", buildErr)
+		zap.L().Error("build failed", zap.Error(buildErr))
+		telemetry.ReportCriticalError(ctx, buildErr)
+
+		dbErr := a.db.EnvBuildSetStatus(ctx, templateID, buildUUID, envbuild.StatusFailed)
+		if dbErr != nil {
+			telemetry.ReportCriticalError(ctx, fmt.Errorf("error when setting build status: %w", dbErr))
+		}
 
 		return
 	}
 
-	// Set the build status to building
-	err = a.db.EnvBuildSetStatus(ctx, envDB.ID, buildUUID, envbuild.StatusBuilding)
+	// status building must be set after build is triggered because then
+	// it's possible build status job will be triggered before build cache on template manager is created and build will fail
+	err = a.db.EnvBuildSetStatus(ctx, templateID, buildUUID, envbuild.StatusBuilding)
 	if err != nil {
 		err = fmt.Errorf("error when setting build status: %w", err)
 		telemetry.ReportCriticalError(ctx, err)
-
-		a.buildCache.Delete(templateID, buildUUID, team.ID)
-
-		return
 	}
 
-	// Trigger the build in the background
-	go func() {
-		buildContext, childSpan := a.Tracer.Start(
-			trace.ContextWithSpanContext(context.Background(), span.SpanContext()),
-			"background-build-env",
-		)
-		defer childSpan.End()
-
-		startTime := time.Now()
-		build := envDB.Edges.Builds[0]
-		startCmd := ""
-		if build.StartCmd != nil {
-			startCmd = *build.StartCmd
-		}
+	telemetry.ReportEvent(ctx, "created new environment", attribute.String("env.id", templateID))
 
-		// Call the Template Manager to build the environment
-		buildErr := a.templateManager.CreateTemplate(
-			a.Tracer,
-			buildContext,
-			a.db,
-			a.buildCache,
-			templateID,
-			buildUUID,
-			build.KernelVersion,
-			build.FirecrackerVersion,
-			startCmd,
-			build.Vcpu,
-			build.FreeDiskSizeMB,
-			build.RAMMB,
-		)
-		if buildErr != nil {
-			buildErr = fmt.Errorf("error when building env: %w", buildErr)
-			zap.L().Error("build failed", zap.Error(buildErr))
-			telemetry.ReportCriticalError(buildContext, buildErr)
-
-			dbErr := a.db.EnvBuildSetStatus(buildContext, templateID, buildUUID, envbuild.StatusFailed)
-			if dbErr != nil {
-				telemetry.ReportCriticalError(buildContext, fmt.Errorf("error when setting build status: %w", dbErr))
-			}
-
-			// Save the error in the logs
-			buildCacheErr := a.buildCache.Append(templateID, buildUUID, fmt.Sprintf("Build failed: %s\n", buildErr))
-			if buildCacheErr != nil {
-				telemetry.ReportCriticalError(buildContext, fmt.Errorf("error when appending build logs: %w", buildCacheErr))
-			}
-
-			cacheErr := a.buildCache.SetDone(templateID, buildUUID, api.TemplateBuildStatusError)
-			if cacheErr != nil {
-				telemetry.ReportCriticalError(buildContext, fmt.Errorf("error when setting build done in logs: %w", cacheErr))
-			}
-
-			return
-		}
+	// Do not wait for global build sync trigger it immediately
+	go func() {
+		a.templateManager.BuildStatusSync(context.Background(), buildUUID, templateID)
```

Comment:
i am still missing the background context we used before (I mentioned it on the other place you were using it)

Reply:
done, so span is extended into child